dwio/nimble/common/Types.h

```diff
@@ -120,6 +122,7 @@ enum class DataType : uint8_t {
 
 std::string toString(DataType dataType);
 std::ostream& operator<<(std::ostream& out, DataType dataType);
+fmt::underlying_t<DataType> format_as(DataType dataType);
```

Comment:
This is needed for `VELOX_CHECK_EQ()`: https://github.com/facebookincubator/nimble/blob/81afca4f435a40bfdfd911dec5c682f2bd9a5d2e/dwio/nimble/velox/selective/ChunkedDecoder.cpp#L143

`VELOX_CHECK_EQ()` uses `fmt`. So we need this to format `DataType` values used for `VELOX_CHECK_EQ()`.